### PR TITLE
Paper: Programmatically Identifying Cognitive Biases Present in Software Development

### PR DESCRIPTION
**Pull request by mwidjaja1**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

*Editor*: @deniederhut

*Reviewers*: @tskluzac

### CODE REVIEW

**Comment by mwidjaja1** [comment]
For those reviewing this paper -- first off, thanks! We appreciate everyone taking the time to review this paper.

As a full confession, I am personally new to using .rst and .bib file formats in general. Any feedback on how I could leverage those formats better would be much appreciated. Of specific note, I struggled to place my tables + plots in a way that wouldn't get auto-written by the Scipy build script. Any advice there would be greatly appreciated.

Also of note for reviewers, the corporation I work for has an incredibly strict review process. Any revisions that would be substantial to the paper's contents would require a re-review by several departments, which could take 3+ weeks. So if you have a request that would significantly change the contents of our paper, it'd be much appreciated if we can begin that conversation sooner rather than later, to respect the Scipy deadlines as best as we can.

Many thanks again to all reviewing, we're excited to contribute to the community.

**Comment by mwidjaja1** [comment]
For those reviewing this paper -- first off, thanks! We appreciate everyone taking the time to review this paper.

As a full confession, I am personally new to using .rst and .bib file formats in general. Any feedback on how I could leverage those formats better would be much appreciated. Of specific note, I struggled to place my tables + plots in a way that wouldn't get auto-written by the Scipy build script. Any advice there would be greatly appreciated.

Also of note for reviewers, the corporation I work for has an incredibly strict review process. Any revisions that would be substantial to the paper's contents would require a re-review by several departments, which could take 2-3 weeks. So if you have a request that would significantly change the contents of our paper, it'd be much appreciated if we can begin that conversation sooner rather than later, to respect the Scipy deadlines as best as we can.

Many thanks again to all reviewing, we're excited to contribute to the community.

**Comment by deniederhut** [comment]
> As a full confession, I am personally new to using .rst and .bib file formats in general.

No worries! We're here to help.

> I struggled to place my tables + plots in a way that wouldn't get auto-written by the Scipy build script

I'm not sure what you mean by this. Could you help me understand what you expected / wanted to happen, and how that differs from the build outputs?

**Comment by mwidjaja1** [comment]
Thanks @deniederhut! On the PDF of our paper (https://procbuild.scipy.org/download/mwidjaja1-2021)

- Ideally, Page 3-4 would have Figure 4 & Figure 5, in addition to Table 2, Table 3, Figure 3 which they already have. That's because those pages correspond to Figure 4 & 5.
- That previous request, certainly, would push some content from Page 4 onto Page 5. And that's ok with us because then, we'd move Table 5 to Page 5, where it'd be closer to where the Modeling section got pushed to.

But, unless I'm being silly here, I don't think I see a way to move Figure 4 & 5 to Page 4. When I try moving it in the .rst file, when I run the build script, it places Figures 4 & 5 back on Page 5. At least I think that's what happens -- I could be being silly with the .rst file.

Does that make sense? Any thoughts? Many thanks!

**Comment by deniederhut** [comment]
@scoobies assign @deniederhut as editor

**Comment by scoobies** [comment]
Assigned! @deniederhut is now the editor

**Comment by deniederhut** [comment]
I understand. So we use tex to compose the layout of these documents, and it has a series of heuristics it uses to decide what should go where. Generally, we consider this to be a feature (thanks Knuth!), because typesetting is really hard, but there are [ways to force parts of the document to appear in specific places](https://tex.stackexchange.com/questions/8625/force-figure-placement-in-text). This gets a little bit more complicated because we create the latex source dynamically by transpiling the rst inputs, but we could probably make this work if we really wanted to with raw directives.

So, to be brief, you are not being silly -- the typesetters ignore the location of your tables and images in the source file, and rely on captions and references to those captions to link text to tables and figures. How strongly do you feel about changing this behavior?

**Comment by mwidjaja1** [comment]
Hey @deniederhut, thanks for that great explanation. That makes a ton of sense and it sounds like quite the challenge!

In an ideal utopia, it'd be nice to at least provide a preference on which PDF page an image/table should be on. But in this world, I don't feel strongly about changing that behavior. I'm sure there's more under the hood than I realize and I trust the team that they've made the best choices in light of what's available. I just feel better now that it wasn't my naiveness on why I couldn't get the images where I thought they would have been.

Thanks!

**Comment by deniederhut** [comment]
Of course! Let me know if you change your mind and we'll see if you and I can hack something together.

**Comment by mwidjaja1** [comment]
Thanks @deniederhut! Actually I had another question, I'm not sure if you're the person to ask, but I'm hoping you'd know who I would ask if it's not you.

We are planning to present a poster at Scipy. Right now our 'poster' is in the form of 5-6 PowerPoint Slides -- we felt that keeping it to the slide format, even though it was a poster, helped communicate a stronger story. It also makes our legal review team much happier and less intimidated over say, giving them a Jupyter Notebook or Interactive Web App. Does Scipy have any best practices and/or preferences on how we share this PowerPoint deck as our 'poster'? For example,

- Format: Does the team prefer just a pure PDF with the slides or like a Markdown page with the slides embedded in them and summary text prior to each slide?
- Paper Links: I assume it's a 'good' practice to link out to this paper as part of our poster. Do you know what the 'final' URL to the paper will be?
- Anything else we should be aware about and/or may want to change?

Thanks!

**Comment by deniederhut** [comment]
> Does Scipy have any best practices and/or preferences on how we share this PowerPoint deck as our 'poster'?

🤷‍♀️ I think we're still figuring out this online format too. You might want to reach out to the program chairs (Madicken and Matt) to see if they have some ideas about good formats.

What I can say is that the conference publishes posters/slides alongside the full papers (see e.g. http://conference.scipy.org/proceedings/scipy2020/slides.html) -- we archive them and submit them to crossref so they're citable. We'll announce that during the conference (most people don't have their posters ready this far in advance!), but you're welcome to open a PR for your slide deck now if you want to. Those instructions are here: https://github.com/scipy-conference/scipy_proceedings#instructions-for-slides.

**Comment by mwidjaja1** [comment]
Thanks @deniederhut! Our poster is 'ready' but not that ready, we still have to send it through our review board (and who knows how long that'll take). But, thanks for your help in pointing us to the right people so we can take next steps to ensure our review submission is as close to what everybody wants.

**Comment by deniederhut** [comment]
@scoobies add @tskluzac to reviewers

**Comment by scoobies** [comment]
@tskluzac added to the reviewers list!

**Comment by deniederhut** [comment]
We've been having difficulty finding reviewers for some papers, so to make sure everyone has enough time to do their reviews, we will be bumping the deadline for the initial review back to June 17th.

**Comment by tskluzac** [comment]
Hi all, 

Hi all, I'm excited to be a reviewer for "Programmatically Identifying Cognitive Biases Present in Software Development"!  To quickly introduce myself -- my name is Tyler Skluzacek, and I'm a grad student in Computer Science at UChicago. I'll give this paper a read tomorrow, and get provide my initial thoughts in the next 1-2 days. 

Best, 
Tyler

**Comment by mwidjaja1** [comment]
Hey thanks @tskluzac! Looking forward for your feedback!

To give you a heads up, the corporation we work for has a pretty stringent review legal process unfortunately. If any of your proposed changes would require substantial content additions to the paper, we would need to modify our paper and then re-submit our paper to our legal team by Thursday June 10th end of business day, to have any shot at meeting Scipy's deadlines. 

So if you do have any recommendations for substantial content additions, it'd be much appreciated if you could let us know sooner rather than later.

Thanks!

**Comment by deniederhut** [comment]
> by Thursday June 10th end of business day, to have any shot at meeting Scipy's deadlines.

I'm not sure which deadline this is referring to, but we won't publish the finalized conference proceedings until the end of July. I'd like to reiterate that the official deadline for the initial review is 17th June.

**Comment by mwidjaja1** [comment]
@deniederhut I must have sworn there was a deadline for the paper after the final reviews are up on around June 24th. But if you didn't hear about that due date, that I'm going to pretend it doesn't exist and live happier 🙂

**Comment by deniederhut** [comment]
Ah, yes. So we publish a "pre-conference" version of proceedings in advance of the conference, so people can look through the papers. We have a hard deadline of the conference start date, so we ask authors to have their edits finished a week or so in advance. 

The "official" version includes talk slides and posters, and it takes a while to collect those and get them archived, so the final version goes out toward the end of July. I'm happy to keep in touch with you to make sure your submission makes it through compliance before this happens.

**Comment by tskluzac** [comment]
Hello authors! 

I had the chance to read this paper, and I'll try my best to avoid bias in this review (ha). This paper certainly has a fascinating premise in that the ways we develop software implicitly contain bias, and these biases are realized in the form of commit messages and comments. I'll start with the things I thought were particularly strong: 

- the study itself. It seems the authors had a robust plan in place to ensure agreement across multiple reviewers. The diagram on page 3 was particularly helpful to my understanding. 
- the provided examples. The bias is subtle. I'm glad you guys gave examples of comments containing the different types of bias. I'll admit I scratched my chin for a while on some of them while trying to confirm the bias, and I ultimately agreed in each case. 
- the broad visualization of bias results, particularly figures 4 and 5. I like how you subdivided the analysis both across projects (codebases) AND across the type of comment (docstring, comment, commit message). I for one know that I'm more volatile in commit messages, as many of them are written in frustration. 

In my next message (in the next hour or so)  I will outline room for improvement. I do not think any of the changes will require re-running any of your studies, so you should be fine from an administrative/company standpoint. If they are, just let me know.

**Comment by tskluzac** [comment]
The following are the major improvement items that would immediately improve the quality of the paper: 

- Information about reviewers: in a paper about bias, we should be cognizant as to how the reviewers themselves can be biased. Were they the ones who worked on these codebases? What were the race and sex of the reviewers? I think more information about the reviewers is important to full appreciating the review process. 
- Writing clarity: this paper can better focus on key ideas by minor reorganization *and* by condensing language throughout. Each subsection within a section seems to be a surprise. Can you, at the start of the section (before subsections) provide a broad overview of the section and introduce the ordering of the subsections? For instance this could be structured as "Overarching statement introducing the point of the section >> list of the following subsections (across one sentence or multiple) and how they tie back to that overarching statement". Then I would just be sure to link the ideas of paragraphs from one to the next >> for instance "To validate models..." is followed by "Python scripts were developed..." --> these paragraphs are largely unrelated and I'm not sure what story you're trying to tell. Each paragraph should be structured as "claim --> description/data --> wrap-up statement"; for instance the paragraphs that begin with "To validate models" and "Python script were developed", these largely just describe research tasks without any context (claim) or conclusion about what your tasks accomplished (wrap-up statement). 
- Captions should be descriptive. As standalone entities, I should understand the main idea of each graph and table. Give me a couple sentences in each caption so  the reader does not need to flip between the text and the graph as often. For instance in Table 2: I have no idea what "Answer" "Annotation" ... "Bias" tags mean without reading the text. Or what the +/- is for. Or whether 0.85 is a percentage. 
- The use of weak verbs and passive voice throughout makes some sentences unnecessarily wordy (which was sometimes distracting from the content of the work). "There are hundreds of biases that have been described across fields of research" could be "Researchers across multiple fields describe biases...". The subject [Researchers across multiple fields] strong verbed [describe] ... " Try your best to shorten sentences by removing unnecessary words. 
- Evaluation debrief: you talk a lot about the figures in the evaluation. What do these figures mean with regards to the larger questions of bias? 

The following are minor items: 
- Fig 3 has error bars that stretch above 100%, which doesn't make intuitive sense. Maybe a better method would be an interquartile range (box) plot? 
- "No models were trained for the dataset composed of docstrings, due to only 7 (of 181) entries being labeled as biased." --> this seems like an important finding that shouldn't be hidden away in the text! Docstrings are inherently unbiased, likely due to them being highly technical/methodical in nature.

**Comment by tskluzac** [comment]
Overall, a super fun and insightful paper. Let me know if you have any questions, or would like any clarifications! I'd also be happy to have another look if needed.

**Comment by deniederhut** [comment]
@scoobies mark needs more review

**Comment by mwidjaja1** [comment]
Hey thanks @tskluzac for your feedback! It's much appreciated and we're glad it was a fun paper. I had one follow-up question for you and one heads up.

First, could you elaborate on your feedback: "_Evaluation debrief: you talk a lot about the figures in the evaluation. What do these figures mean with regards to the larger questions of bias_". The overwhelming majority of our figures were to visualize the dataset we annotated and modeled off of, to give situational awareness for what biases were in our dataset. This is important during modeling because we want to be aware of any data in-balance we might have during modeling and work around those. Did you want us to elaborate more on that or did I miss your question?

And as a heads up, you are right, almost all of your feedback shouldn't require a re-review. However one of your points could require a re-approval since its technically not information that we told our team we'd share (even if it's silly in the grand scheme of things). I'm running that by the board right now as we speak to see if that does require a whole re-review, and if it does, I'll let you know. 

Many thanks!

**Comment by tskluzac** [comment]
Thanks for your reply @mwidjaja1! 

In regards to the "Evaluation debrief" point, I feel like the results section just says 'we have x reliability, y F1' -- but there's no actual discussion around the findings themselves (i.e., the graphs). I see there's an entire half page of graphs that have rainbow colors such that each color represents a type of bias, but what is the takeaway here? Why is 'Other' the leading the way in some cases? Why are some of the reviewers so non-correlated with one another? Even if there's no definitive answer, some takeaways from the graphs should be given. 

Let me know if that sounds reasonable!

**Comment by mwidjaja1** [comment]
Hey @tskluzac. So I'm following your point though I think I see what's going on here too. In short: None of the figures talk about model results. They're all about the input data for the models and either how (1) human reviewers tended to disagree on specific types of biases or (2) what those human reviewers decided upon as the type of bias.

So to answer your question specifically: The answers you're probably looking for are in the "Results & Discussion: Annotated Datasets" & "Conclusions & Implications" sub-section. We described that 'Other' are all the biases we found but are not one of the four bias types we originally selected. Our suspected reasoning on why some reviewers found more 'Other' biases than other reviewers, ties back to our theory of why we think reviewers tended to disagree on the particular types of bias in the Conclusions & Implications section.

But to circle back to what I think the source of confusion is, this actually began with a conversation I had with @deniederhut at https://github.com/scipy-conference/scipy_proceedings/pull/621#issuecomment-848339193 about the paper generation script. Because it lays out images/tables automatically, we were unable to group similar figures/tables together. We originally wanted to:

- Table 1-2 & Figure 3 should have been linked together, for the first part of the 'Results & Discussion: Annotated Datasets' section 
- Table 3 & Figure 4-5 should have been linked together, for the second part of the 'Results & Discussion: Annotated Datasets' section 
- Table 4 is the only table for model results (a little bit more why in a sec)

Seeing how these figures and tables are a bit all over the place, it seems like we may have accidentally confused you into thinking they're model results. Unless there's some other easy way to fix the layout, I can take an action to revise the captions of those figures + tables some more, to describe how they're visualizing the input data the models are getting. The unclear captions were definitely our fault, so our apologies for that.

Back to Table 4, it is somewhat intentional that it became the only artifact of the model results. When the paper was getting reviewed by our review board, our plots and more data confused our review board quite a bit, because there were so many of them and they seemed haphazardly injected. In light of how confused they got, we made an executive decision of sorts that because our final models were basic 'binary' models which only predicted for the presence of bias (not the specific bias label persay), providing a summary table would be less confusing and informative enough, versus the original plots we provided. Of course, hindsight is 20/20 and looking at your feedback plus our review team's feedback, we probably should have split the difference and provided a plot to describe the last 'comprehensive' model and a table to summarize the other models we made.

So to summarize next steps we could take,

1. I'll definitely go back to update the captions for the figures to make it clear that they describe the dataset the models got and NOT model results.
2. I can ask again to see if we can include the confusion matrices for the final 'comprehensive' model to expand on the model results. However the review board informed me if we were to do this, it'd take at least another 3-4 weeks since they're jammed up at the moment. Not trying to discourage it persay because I agree we did fly by the model results a bit too quickly, I just want to be transparent about what it'll take to fix it as unfortunate as it might be :/. 

What do you think? Thanks!

**Comment by deniederhut** [comment]
> Table 3 & Figure 4-5 should have been linked together, for the second part of the 'Results & Discussion: Annotated Datasets' section

Hm... another option might be to combine the two figures into a single image, and have the caption for figure 4a and figure 4b?

**Comment by mwidjaja1** [comment]
> > Table 3 & Figure 4-5 should have been linked together, for the second part of the 'Results & Discussion: Annotated Datasets' section
> 
> Hm... another option might be to combine the two figures into a single image, and have the caption for figure 4a and figure 4b?

That's certainly an idea we can play around with! We were less concerned about how the figures are separated, and more focused on wanting those figures to visually appear before Table 4. Right now those two figures live with Table 4, which makes it seem like they're model results even though they're aren't. We'll give that a try though to see if that helps the layout in that area!

Many thanks!

**Comment by tskluzac** [comment]
> Hey @tskluzac. So I'm following your point though I think I see what's going on here too. In short: None of the figures talk about model results. They're all about the input data for the models and either how (1) human reviewers tended to disagree on specific types of biases or (2) what those human reviewers decided upon as the type of bias.
> 
> So to answer your question specifically: The answers you're probably looking for are in the "Results & Discussion: Annotated Datasets" & "Conclusions & Implications" sub-section. We described that 'Other' are all the biases we found but are not one of the four bias types we originally selected. Our suspected reasoning on why some reviewers found more 'Other' biases than other reviewers, ties back to our theory of why we think reviewers tended to disagree on the particular types of bias in the Conclusions & Implications section.

Okay, so now I'm understanding regarding the formatting issues on the graphs, which it seems like you're resolving with the help of @deniederhut. 

> But to circle back to what I think the source of confusion is, this actually began with a conversation I had with @deniederhut at [#621 (comment)](https://github.com/scipy-conference/scipy_proceedings/pull/621#issuecomment-848339193) about the paper generation script. Because it lays out images/tables automatically, we were unable to group similar figures/tables together. We originally wanted to:
> 
> * Table 1-2 & Figure 3 should have been linked together, for the first part of the 'Results & Discussion: Annotated Datasets' section
> * Table 3 & Figure 4-5 should have been linked together, for the second part of the 'Results & Discussion: Annotated Datasets' section
> * Table 4 is the only table for model results (a little bit more why in a sec)
> 
> Seeing how these figures and tables are a bit all over the place, it seems like we may have accidentally confused you into thinking they're model results. Unless there's some other easy way to fix the layout, I can take an action to revise the captions of those figures + tables some more, to describe how they're visualizing the input data the models are getting. The unclear captions were definitely our fault, so our apologies for that.

I think descriptive captions would be a sufficient band-aid for clarifying what each graphic is meant to represent. I'm not too up-to-date on what you've been trying, but could you turn Fig. 4 into three separate figures: (1) Codesmell, (2) Project A, (3) Project B -- such that (3) has three graphs in it. I know sometimes LaTeX has an easier time placing small figures than big figures. If you do that, you could also try putting all three graphs for B side-by-side over both columns.

> Back to Table 4, it is somewhat intentional that it became the only artifact of the model results. When the paper was getting reviewed by our review board, our plots and more data confused our review board quite a bit, because there were so many of them and they seemed haphazardly injected. In light of how confused they got, we made an executive decision of sorts that because our final models were basic 'binary' models which only predicted for the presence of bias (not the specific bias label persay), providing a summary table would be less confusing and informative enough, versus the original plots we provided. Of course, hindsight is 20/20 and looking at your feedback plus our review team's feedback, we probably should have split the difference and provided a plot to describe the last 'comprehensive' model and a table to summarize the other models we made.

Yes. I feel like there is some larger wrap-up statement missing, so maybe it should be the first thing you mention in the conclusion? Right now the primary takeaway seems like "we made two well-curated datasets" which is GREAT, but it's not your primary finding. 

> So to summarize next steps we could take,
> 
> 1. I'll definitely go back to update the captions for the figures to make it clear that they describe the dataset the models got and NOT model results.

Awesome! 

> 2. I can ask again to see if we can include the confusion matrices for the final 'comprehensive' model to expand on the model results. However the review board informed me if we were to do this, it'd take at least another 3-4 weeks since they're jammed up at the moment. Not trying to discourage it persay because I agree we did fly by the model results a bit too quickly, I just want to be transparent about what it'll take to fix it as unfortunate as it might be :/.
>
> What do you think? Thanks!

Likely worth a shot, but even more explanation / wrap-up as part of the writing will be a step in the right direction. 

Let me know if you have any more questions!

**Comment by mwidjaja1** [comment]
Hey @tskluzac, once again many thanks for your feedback and understanding in all this.

We have worked on enhancing the captions, cleaning up the wording, and adding a quick blurb about the demographics of the reviewers. All that is ready to go and does not need a re-review. We did experiment with the plot layout, we don't think we nailed it quite yet, but we'll keep poking at it.

What we haven't done yet is elaborate more on the model results and go deeper into the specifics. I still need to check if we're allowed to spend time to work on that. If we are allowed and if we proceed, we'd have to get the paper re-reviewed so it wouldn't be ready until ~July 9th. Is that too late for the team? If not we can definitely consider this option. EDIT AFTER THE FACT: Also if we go down this route of re-review, is there anything else you'd like us to add to the paper? If we're going down this route, we may as well profit 🙂 

Once again many thanks for all this. It's great getting feedback from someone outside our bubble.

**Comment by mwidjaja1** [comment]
Hey all,

As a quick update, to keep the ball rolling, we've went ahead with making the edits and plan to submit it to the review board again. If we change our minds due to deadlines and what not, it's not a problem at all, we can take it back from the board and remove the less questionable updates. 

The big additions we've added to our paper are:

- Why the Multi-Label models didn't work despite having a seemingly good F1
- A plot to show exactly what happened in our 'best' Binary Model
- Demographic information
- Some more explanation about one of our more 'unique' hyperparameters
- And of course, enhanced captions everywhere.

 If there's anything else in the wish list, now is the time! If not, I plan to ship it off for review around noon EST today. As always, many thanks for your feedback.

**Comment by deniederhut** [comment]
@mwidjaja1 to give everyone more time to respond to their reviews, we've pushed back the deadline for authors' final revisions to July 1st

**Comment by mwidjaja1** [comment]
> @mwidjaja1 to give everyone more time to respond to their reviews, we've pushed back the deadline for authors' final revisions to July 1st

Thanks! I've asked our review board to speed track this paper revision since it's just a revision. They're trying their darnest to get the paper reviewed by July 2nd, but given how we don't work on Fridays, that's likely to be on July 1st itself. 

Many thanks again for the feedback and your understanding with all the back and forth.

**Comment by mwidjaja1** [comment]
@deniederhut I think I goofed up and accidentally uploaded our poster to this same MR. Would you like me to undo it or is it ok for the poster to track on with this MR since it is similar content with the paper.

Sorry about that!

**Comment by deniederhut** [comment]
No worries! We can make that work

**Comment by deniederhut** [comment]
@scoobies check references

**Comment by scoobies** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1136/jech.2003.008466 is OK
- 10.1109/TSE.2018.2877759 is OK
- 10.1145/203241.203256 is OK
- 10.1007/s11219-014-9250-6 is OK
- 10.1145/1378727.1389970 is OK
- 10.1109/IWoR.2019.00017 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

File: papers/lm_cognitive/IEEEtran.cls [hunk 1]
```diff
@@ -0,0 +1,6347 @@
+%%
```
Comment (deniederhut) [inline papers/lm_cognitive/IEEEtran.cls]: Did you mean to commit your own stylesheet here?

**Comment by deniederhut** [comment]
@scoobies mark pending comment

File: papers/lm_cognitive/IEEEtran.cls (hunk 1 above)
Comment (mwidjaja1) [reply papers/lm_cognitive/IEEEtran.cls]: So uhh [whispers to you] I actually have no idea what this file is, I just noticed it somewhere lurking in one of the Scipy Best Practice Repos or a website that described how to convert our files to this format. If you're implying this file isn't necessary then I'm more than inclined to believe you because I don't know what I'm doing with this file.

Comment (deniederhut) [reply papers/lm_cognitive/IEEEtran.cls]: Haha no worries! We supply our own stylesheet for the conference papers, so you shouldn't need one.

Comment (mwidjaja1) [reply papers/lm_cognitive/IEEEtran.cls]: Oh sweet! We'll delete it then.

**Comment by deniederhut** [comment]
Awesome -- thanks for making that change @mwidjaja1 .

@tskluzac , do you feel this paper is now ready for inclusion in the proceedings?

**Comment by mwidjaja1** [comment]
> Awesome -- thanks for making that change @mwidjaja1 .
> 
> @tskluzac , do you feel this paper is now ready for inclusion in the proceedings?

He probably doesn't yet because we haven't given him the revision yet (and if he does approve, well that was easy! 🙂).

That paper is still going through the review process and I'm pushing the review team for a completion either today or tomorrow, July 1st. The moment it's approved, I'll upload the PDF here. Sorry for the delay!

**Comment by deniederhut** [comment]
Ah that's right. Sorry -- it can be hard to keep track of where all the submissions are

**Comment by mwidjaja1** [comment]
@tskluzac and @deniederhut attached should be the revisions of our paper, based on the feedback, that have been approved by our review team.

We welcome all feedback like before but as a reminder, we'd be unable to introduce any new content (which includes analysis/conclusions based on the content) without going through that review process again. Thanks for your patience!

**Comment by deniederhut** [comment]
Awesome! @tskluzac , do you feel this paper is now ready for inclusion in the proceedings?

**Comment by tskluzac** [comment]
@mwidjaja1 @deniederhut  Where's the PDF? I'm not seeing it attached here.

**Comment by tskluzac** [comment]
Found it here! Reviewing now
https://procbuild.scipy.org/download/mwidjaja1-2021

**Comment by tskluzac** [comment]
@mwidjaja1:  Awesome updates! I feel the explanations of the process and the models are much easier to follow, the graphics are much more self-explainable, and the confusion matrices (and the new conclusion) helped me to understand the results and implications of the experiment. Thank you for your hard work on this these past few weeks!

@deniederhut: I am happy with the updates, and would like to see this paper included in the SciPy21 proceedings.

**Comment by mwidjaja1** [comment]
> @mwidjaja1: Awesome updates! I feel the explanations of the process and the models are much easier to follow, the graphics are much more self-explainable, and the confusion matrices (and the new conclusion) helped me to understand the results and implications of the experiment. Thank you for your hard work on this these past few weeks!
> 
> @deniederhut: I am happy with the updates, and would like to see this paper included in the SciPy21 proceedings.

Hey thanks so much @tskluzac for your feedback too. It’s not too often we get to have external feedback on our ideas so your feedback was much appreciated. Excited to share it with the community?

@deniederhut do you need anything else from me to proceed? Of note, anything about our poster which technically is part of this MR too?

**Comment by deniederhut** [comment]
Mmm nope I think we're good to go here. Thanks everyone!

**Comment by deniederhut** [comment]
@scoobies mark ready
